tests/test_json_schema.py

```diff
@@ -1641,6 +1641,25 @@ class Baz(BaseModel):
         models_json_schema([(Bar, 'validation'), (Baz, 'validation')], ref_template='/schemas/{bad_name}.json#/')
 
 
+def test_external_ref():
+    """https://github.com/pydantic/pydantic/issues/9783"""
+
+    class Model(BaseModel):
+        json_schema: typing.Annotated[
+            dict,
+            WithJsonSchema({'$ref': 'https://json-schema.org/draft/2020-12/schema'}),
+        ]
+
+    assert Model.model_json_schema() == {
+        'properties': {
+            'json_schema': {'allOf': [{'$ref': 'https://json-schema.org/draft/2020-12/schema'}], 'title': 'Json Schema'}
```

Comment:
Hmm, if it's an external ref, I wonder if we want to keep the other schema info?

Reply:
I'm wondering why there's an `allOf` rule here since there's only one entry. In general shouldn't `WithJsonSchema` just be replacing the JSON Schema for that attribute? Is it this way because it's trying to keep the `title` attribute?

Reply:
See https://github.com/pydantic/pydantic/pull/10029, that should fix this (I think)

Reply:
> Hmm, if it's an external ref, I wonder if we want to keep the other schema info?

I think this is actually a consequence of the `WithJsonSchema` annotation, so isn't a problem sourced from this PR.